kubespawner/clients.py

```diff
@@ -41,8 +43,16 @@ def shared_client(ClientType, *args, **kwargs):
         # Kubernetes client configuration is handled globally
         # in kubernetes.py and is already called in spawner.py
         # or proxy.py prior to a shared_client being instantiated
-        Client = getattr(kubernetes.client, ClientType)
+        Client = getattr(kubernetes_asyncio.client, ClientType)
         client = Client(*args, **kwargs)
         # cache weakref so that clients can be garbage collected
         _client_cache[cache_key] = weakref.ref(client)
+
     return client
+
+
+async def load_config():
+    try:
+        kubernetes_asyncio.config.load_incluster_config()
+    except kubernetes_asyncio.config.ConfigException:
+        await kubernetes_asyncio.config.load_kube_config()
```

Comment:
You extracted parts of the load_config logic into its dedicated function, I figure we should extract all parts there. I understand that would mean we need to pass some configuration but thinks that the right call.

It makes sense for such logic to be here, but not for it to be directly under initialize resources or a resource named like that.

Reply:
Will do.

Reply:
Done.  I'm not a huge fan of how I did it but short of refactoring it into a mixin class that Spawner, Proxy, and Reflector all used, I don't immediately see a better way than having the load_config requestor pass a reference to itself into load_config so load_config can check for the attributes and use them if it finds them.